src/others/file_explorer/lv_file_explorer.c

```diff
@@ -0,0 +1,782 @@
+    lv_obj_align(explorer->quick_access_ctrl_btn, LV_ALIGN_LEFT_MID, 0, 0);
+    lv_obj_add_flag(explorer->quick_access_ctrl_btn, LV_OBJ_FLAG_CHECKABLE);
+
+    // Label on the quick access control button
+    lv_obj_t * label = lv_label_create(explorer->quick_access_ctrl_btn);
+    lv_label_set_text(label, LV_SYMBOL_LIST);
+
+    // Two lists of quick access bar
+    lv_obj_t * btn;
+    // list 1
+    explorer->list_device = lv_list_create(explorer->quick_access_area);
+    lv_obj_set_size(explorer->list_device, LV_PCT(100), LV_SIZE_CONTENT);
+    lv_obj_set_style_bg_color(lv_list_add_text(explorer->list_device, "DEVICE"), lv_palette_main(LV_PALETTE_ORANGE), 0);
+
+    btn = lv_list_add_btn(explorer->list_device, NULL, LV_SYMBOL_DRIVE " File System");
+    lv_obj_add_event_cb(btn, quick_access_event_handler, LV_EVENT_CLICKED, obj);
+    btn = lv_list_add_btn(explorer->list_device, NULL, LV_SYMBOL_DRIVE " Mnt");
+    lv_obj_add_event_cb(btn, quick_access_event_handler, LV_EVENT_CLICKED, obj);
+
+    // list 2
+    explorer->list_places = lv_list_create(explorer->quick_access_area);
+    lv_obj_set_size(explorer->list_places, LV_PCT(100), LV_SIZE_CONTENT);
+    lv_obj_set_style_bg_color(lv_list_add_text(explorer->list_places, "PLACES"), lv_palette_main(LV_PALETTE_LIME), 0);
+
+    btn = lv_list_add_btn(explorer->list_places, NULL, LV_SYMBOL_HOME " HOME");
+    lv_obj_add_event_cb(btn, quick_access_event_handler, LV_EVENT_CLICKED, obj);
+    btn = lv_list_add_btn(explorer->list_places, NULL, LV_SYMBOL_VIDEO " Video");
+    lv_obj_add_event_cb(btn, quick_access_event_handler, LV_EVENT_CLICKED, obj);
+    btn = lv_list_add_btn(explorer->list_places, NULL, LV_SYMBOL_IMAGE " Pictures");
+    lv_obj_add_event_cb(btn, quick_access_event_handler, LV_EVENT_CLICKED, obj);
+    btn = lv_list_add_btn(explorer->list_places, NULL, LV_SYMBOL_AUDIO " Music");
+    lv_obj_add_event_cb(btn, quick_access_event_handler, LV_EVENT_CLICKED, obj);
+    btn = lv_list_add_btn(explorer->list_places, NULL, LV_SYMBOL_FILE "  Documents");
+    lv_obj_add_event_cb(btn, quick_access_event_handler, LV_EVENT_CLICKED, obj);
+#endif
+
+    // Show current path
+    explorer->path_label = lv_label_create(explorer->head_area);
+    lv_label_set_text(explorer->path_label, LV_SYMBOL_EYE_OPEN"https://lvgl.io");
+    lv_obj_center(explorer->path_label);
+
+    // Table showing the contents of the table of contents
+    explorer->file_table = lv_table_create(explorer->browser_area);
+    lv_obj_set_size(explorer->file_table, LV_PCT(100), LV_PCT(86));
+    lv_table_set_col_width(explorer->file_table, 0, LV_PCT(100));
+    lv_table_set_col_cnt(explorer->file_table, 1);
+    lv_obj_add_event_cb(explorer->file_table, browser_file_event_handler, LV_EVENT_ALL, obj);
+    // only scroll up and down
+    lv_obj_set_scroll_dir(explorer->file_table, LV_DIR_TOP | LV_DIR_BOTTOM);
+
+    // Initialize style
+    init_style(obj);
```

Comment:
If `lv_style_init` is called inside of it on every `lv_file_explorer_create`there will be a memory leak as the styles are not reset (freed), just always initialized. 

I suggest having a  `lv_file_explorer_init()` function, called from `lv_init()` and initialize the styles there and here only add the styles to the objects.

Reply:
This is a good suggestion, but I think that in a project, generally only one `file_explorer` needs to be used. How about using **[local styles](https://docs.lvgl.io/master/overview/style.html#local-styles)**?

Reply:
I keep `quick_access_list_btn_style`, which is to prepare for future user expansion.